cranelift/codegen/src/ir/dfg.rs

```diff
@@ -977,6 +957,87 @@ impl DataFlowGraph {
         }
     }
 
+    /// Like `call_signature` but returns none for tail call instructions.
+    fn non_tail_call_signature(&self, inst: Inst) -> Option<SigRef> {
+        let sig = self.call_signature(inst)?;
+        match self.insts[inst].opcode() {
+            ir::Opcode::ReturnCall | ir::Opcode::ReturnCallIndirect => None,
+            _ => Some(sig),
+        }
+    }
+
+    // Only for use by the verifier. Everyone else should just use
+    // `dfg.inst_results(inst).len()`.
+    pub(crate) fn num_expected_results_for_verifier(
```

Comment:
What do you think about inlining this in the verifier? All the fields used are public on the dfg, and there's currently only one call site.

Reply:
Nick and I waffled over that. I kinda felt that it was nice to have it next to `DataFlowGraph::inst_result_types` since they have nearly identical structures. I don't feel super strongly about it though. I suspect there's some other way of factoring all this that would be even better but I haven't been able to figure out what that would be yet...